### PR TITLE
Add server42 service rules rule manager implementation

### PR DESCRIPTION
**Pull request by niksv**

### CODE REVIEW

File: src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/DataAdapter.java [hunk 1]
```diff
@@ -36,4 +38,9 @@
     Switch getSwitch(SwitchId switchId);
 
     SwitchProperties getSwitchProperties(SwitchId switchId);
+
+    KildaFeatureToggles getFeatureToggles();
+
+    Set<Integer> getSwitchIslPorts(SwitchId switchId);
+    // todo s42-generators: implement
```
Comment (rozdy) [inline src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/DataAdapter.java:45]: remove todo

File: src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/ServiceRulesGeneratorFactory.java [hunk 2]
```diff
@@ -228,4 +236,75 @@ public ArpTransitRuleGenerator getArpTransitRuleGenerator() {
...
+                .config(config)
+                .server42Port(server42Port)
+                .islPort(islPort)
+                .build();
+    }
+
+    /**
+     * Get Server 42 ISL RTT turning rule generator.
+     */
+    public Server42IslRttTurningRuleGenerator getServer42IslRttTurningFlowGenerator() {
```
Comment (rozdy) [inline src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/ServiceRulesGeneratorFactory.java:292]: `FlowGenerator` -> `RuleGenerator`

File: src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/ServiceRulesGeneratorFactory.java [hunk 3]
```diff
@@ -228,4 +236,75 @@ public ArpTransitRuleGenerator getArpTransitRuleGenerator() {
...
+        return Server42IslRttTurningRuleGenerator.builder()
+                .config(config)
+                .build();
+    }
+
+    /**
+     * Get Server 42 ISL RTT output rule generator.
+     */
+    public Server42IslRttOutputRuleGenerator getServer42IslRttOutputFlowGenerator(
```
Comment (rozdy) [inline src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/ServiceRulesGeneratorFactory.java:301]: `FlowGenerator` -> `RuleGenerator`

File: src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/RuleManagerImpl.java [hunk 4]
```diff
@@ -121,23 +123,21 @@ public RuleManagerImpl(RuleManagerConfig config) {
...
-        return getServiceRuleGenerators(switchProperties).stream()
+    private List<SpeakerCommandData> buildServiceRules(Switch sw, DataAdapter adapter) {
+        return getServiceRuleGenerators(sw.getSwitchId(), adapter).stream()
                 .flatMap(g -> g.generateCommands(sw).stream())
                 .collect(toList());
     }
 
     @VisibleForTesting
-    List<RuleGenerator> getServiceRuleGenerators(SwitchProperties switchProperties) {
+    List<RuleGenerator> getServiceRuleGenerators(SwitchId switchId, DataAdapter adapter) {
```
Comment (niksv) [inline src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/RuleManagerImpl.java:140]: I changed signature here to do not load ISL from DB if we don't need them. (when RTT is disabled)

**Review by rozdy — APPROVED** [review]

**Review by dpoltavets — APPROVED** [review]

**Pull request merged**